java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightClient.java

```diff
@@ -195,31 +196,29 @@ public ClientStreamListener startPut(FlightDescriptor descriptor, VectorSchemaRo
    * @param root VectorSchemaRoot the root containing data
    * @param metadataListener A handler for metadata messages from the server.
    * @param options RPC-layer hints for this call.
-   * @return ClientStreamListener an interface to control uploading data
+   * @return ClientStreamListener an interface to control uploading data.
+   *     {@link ClientStreamListener#start(VectorSchemaRoot, DictionaryProvider)} will already have been called.
    */
   public ClientStreamListener startPut(FlightDescriptor descriptor, VectorSchemaRoot root, DictionaryProvider provider,
       PutListener metadataListener, CallOption... options) {
-    Preconditions.checkNotNull(descriptor);
-    Preconditions.checkNotNull(root);
+    Preconditions.checkNotNull(descriptor, "descriptor must not be null");
```

Comment:
IMO, the additional null messages don't necessarily add much, but feel free to keep them if you like them i guess.